File: 백준/Silver/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.cc
#include <bits/stdc++.h>
using namespace std;

int gcd(int a, int b) {
    if (a==0) return b;
    return gcd(b%a, a);
}

int lcm(int a, int b) {
    return (a*b) / gcd(a, b);
}

int main() {
	int x, y;
	scanf("%d %d", &x, &y);
	printf("%d\n%d\n", gcd(x, y), lcm(x,y));

	return 0;
}

File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.rb
def gcd(a, b)
    if a == 0 then return b end
    return gcd(b % a, a);
end
    
def lcm(a, b)
    return a*b / gcd(a,b) 
end
    
def solution(n, m)
    [gcd(n,m), lcm(n,m)]
end